src/includes/classes/User.php

```diff
@@ -46,7 +46,8 @@ class User
     * @param string $username
     * @param string $passsword
     */
-    public function __construct($username = null, $password = null)
+    //changed $password to be passed as an empty string, as an exception gets thrown if null is used
+    public function __construct($username = null, $password = "")
```

Comment:
Can the database support a null value for the password? If so, I think it might be better to do a check for null being passed e.g. `$this->password = $password === null ?: password_hash($password, PASSWORD_DEFAULT);`

Reply:
The database does not allow a users password to be null, and it also doesn't for the username either, so it may be better to set `$username` and `$password` to and empty string like this `public function __construct($username = "", $password = "")` instead.

Reply:
Ah I was looking at the old schema, the new one does allow for password to be stored as null, have updated my fork with the changes